// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_outer_classname = "MicroDeviceProto";
option objc_class_prefix = "MDev";

package microDevice;

// The device request service.
service RPCRequest {
  // Send a device serial
  rpc RPCDeviceRequest (DeviceMessage) returns (DeviceReply) {}
  rpc RPCDeviceRelease (DeviceMessage) returns (DeviceReply) {}
  rpc RPCDeviceIsAlive (DeviceMessage) returns (DeviceReply) {}
  rpc RPCSessionRequest (SessionMessage) returns (SessionMessage) {}
  rpc RPCSessionClose (SessionMessage) returns (SessionMessage) {}
  rpc RPCDeviceRequestList (StringMessage) returns (StringMessage) {} 
  rpc RPCDeviceRequestEnable (DeviceMessage) returns (StringMessage) {}
  rpc RPCDeviceRequestDisable (DeviceMessage) returns (StringMessage) {}
  rpc RPCGetDeviceTypeInfo (DeviceMessage) returns (DeviceReply) {}
}

// The device request message
message DeviceMessage {
  string type = 1;
  string serial_number = 2;
  string session_number = 3;
  string user = 4;
}

// The device response message
message DeviceReply {
  string serial_number = 1;
  bool is_alive = 2;
  string vid = 3;
  string pid = 4;
}

message SessionMessage {
  string session_number = 1;
  string task = 2;
}

message StringMessage {
  string text = 1;
}
